lib/filewatch/tail.rb

```diff
@@ -240,8 +240,24 @@ def _sincedb_write
     def quit
       _sincedb_write
       @watch.quit
+      close_files
     end # def quit
 
+    public
+    def close_file(path)
+      @watch.unwatch(path)
+      file = @files.delete(path)
+      return if file.nil?
+      _sincedb_write
+      file.close
+    end
+
+    private
+    def close_files
+      @files.each {|path, file| file.close }
```

Comment:
The body of this method should be moved into `quit`.  `@watch.quit` has done the work of unwatching and the sincedb was written already in so all that is left to do is close the open files and clear their references.